x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/nodes/execute_tools.ts

```diff
@@ -34,23 +35,30 @@ export const executeTools = async ({ config, logger, state, tools }: ExecuteTool
   logger.debug(() => `Node state:\n${JSON.stringify(state, null, 2)}`);
 
   const toolExecutor = new ToolExecutor({ tools });
-  const agentAction = isArray(state.agentOutcome) ? state.agentOutcome[0] : state.agentOutcome;
+  const agentAction = state.agentOutcome;
 
   if (!agentAction || 'returnValues' in agentAction) {
     throw new Error('Agent has not been run yet');
   }
 
-  let out;
-  try {
-    out = await toolExecutor.invoke(agentAction, config);
-  } catch (err) {
-    return {
-      steps: [{ action: agentAction, observation: JSON.stringify(`Error: ${err}`, null, 2) }],
-    };
-  }
+  const steps = await Promise.all(
+    castArray(state.agentOutcome as AgentAction)?.map(async (action) => {
```

Comment:
note: This fixes the multiple tool call code path which was previously not working. So now if the LLM chooses multiple tools at once (only Bedrock right now, OpenAI supports it as well, but not via langchain chatmodel IIRC), both will execute.